artiq/coredevice/phaser.py

```diff
@@ -468,6 +474,27 @@ def get_crc_err(self):
         """
         return self.read8(PHASER_ADDR_CRC_ERR)
 
+    @kernel
+    def measure_frame_timestamp(self):
+        """Measure the timestamp of an arbitrary frame and store it in `self.frame_tstamp`.
+
+        To be used as reference for aligning updates to the FastLink frames.
+        See `get_next_frame_mu()`.
+        """
+        rtio_output(self.channel_base << 8, 0)  # read any register
+        self.frame_tstamp = rtio_input_timestamp(rtio_get_counter() + 125_000, self.channel_base)
```

Comment:
`rtio_counter` is not the right reference here.
`now_mu() + 10*self.t_frame` or something similar.

Reply:
Changed in 392e79c. `4 * t_frame` seems to be enough. Also added an error path in `init()`.

Reply:
Ok. I read from that that you also tested that commit.